File: ocaml/p1.ml
let sumdivisible a max =
  let p = max / a in
  a * (p * (p + 1 )) / 2;;

let () =
  let max = if Array.length Sys.argv > 1 then (int_of_string (Array.get Sys.argv
    1)) - 1 else 999 in
  let sum = sumdivisible 3 max + sumdivisible 5 max - sumdivisible 15 max in
  Printf.printf "%d\n" sum

File: javascript/p1.js
#!/usr/bin/env node

function SumDivisible(n, max){
  let p = Math.floor(max / n);
  return Math.floor(n * (p * ( p + 1 )) / 2);
}
let max = 999;
if (process.argv.length > 2){
  max = process.argv[2]
}
let sum = SumDivisible(3, max) + SumDivisible(5, max) - SumDivisible(15, max);
console.log(sum);
